src/sympc/tensor/replicatedshare_tensor.py

```diff
@@ -157,7 +156,25 @@ def sanity_checks(
         elif len(x.shares) != len(y.shares):
             raise ValueError("Both RSTensors should have equal number of shares.")
 
-        return y
+        rank = 0
+        nr_parties: int = 1  # to handle the case when session_uuid is none.
+        session_uuid = x.session_uuid
+
+        session = None
+        ring_size = None
+        config = None
+
+        if session_uuid is not None:
+            session = sympc.session.get_session(str(x.session_uuid))
+            ring_size = session.ring_size
+            config = session.config
+            rank = session.rank
+            nr_parties = session.nr_parties
+        else:
+            ring_size = x.ring_size
+            config = x.config
+
+        return y, (session, ring_size, config, rank, nr_parties)
```

Comment:
MAYBE: Why don't you simply return the session (in case there is no session you create one with the specific values)

Reply:
Hmmm, but how do we know the number of parties?

Reply:
So this function it is used in the case we are in the MPC setup - we have the `session_uuid` and not in the MPC Setup - the `session_uuid` is `None` and in that scenario, we do not have `session`, right?